### PR TITLE
MM-29170: Add e2e tests to modify teams command

### PR DESCRIPTION
**Pull request by jecepeda**
#### Summary
This PR adds tests to `mmctl team modify` command

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-29170
- Fixes https://github.com/mattermost/mattermost-server/issues/15676

### CODE REVIEW

File: commands/team_e2e_test.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ package commands
 import (
 	"github.com/spf13/cobra"
 
+	"github.com/mattermost/mattermost-server/v5/model"
 	"github.com/mattermost/mmctl/client"
 	"github.com/mattermost/mmctl/printer"
```
Comment (ashishbhate) [inline commands/team_e2e_test.go:11]:
```suggestion
	"github.com/spf13/cobra"

	"github.com/mattermost/mattermost-server/v5/model"
	
	"github.com/mattermost/mmctl/client"
	"github.com/mattermost/mmctl/printer"
```

The goimports lint is complaining. This should fix it.

File: commands/team_e2e_test.go [hunk 2]
```diff
@@ -51,3 +52,48 @@ func (s *MmctlE2ETestSuite) TestRenameTeamCmdF() {
...
+		s.Require().Nil(appErr)
+	})
+	s.Run("user that creates the team can't set team's privacy due to permissions", func() {
+		printer.Clean()
+		teamID := s.th.BasicTeam.Id
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("private", true, "")
+		s.th.LoginBasic()
+		err := modifyTeamsCmdF(s.th.Client, cmd, []string{teamID})
+		s.Require().NoError(err)
```
Comment (ashishbhate) [inline commands/team_e2e_test.go:81]: `modifyTeamsCmdF` won't return an error when the user can't modify the team. Instead it prints the permission error to the terminal. Lets check that the error printed is as expected. 

See https://github.com/mattermost/mmctl/blob/master/commands/user_e2e_test.go#L43-L46 for how to check the printed error text.

File: commands/team_e2e_test.go [hunk 3]
```diff
@@ -51,3 +52,48 @@ func (s *MmctlE2ETestSuite) TestRenameTeamCmdF() {
...
+		s.Require().Equal(model.TEAM_OPEN, t.Type)
+	})
+	s.Run("basic user with normal permissions that hasn't created the team can't set team's privacy", func() {
+		printer.Clean()
+		teamID := s.th.BasicTeam.Id
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("private", true, "")
+		s.th.LoginBasic2()
+		err := modifyTeamsCmdF(s.th.Client, cmd, []string{teamID})
+		s.Require().NoError(err)
```
Comment (ashishbhate) [inline commands/team_e2e_test.go:94]: As above for checking the printed error message.

**Review by ashishbhate — COMMENTED** [review]
Great job @jecepeda ! A few suggestions inline

File: commands/team_e2e_test.go [hunk 4]
```diff
@@ -51,3 +52,48 @@ func (s *MmctlE2ETestSuite) TestRenameTeamCmdF() {
...
+
+func (s *MmctlE2ETestSuite) TestModifyTeamsCmdF() {
+	s.SetupTestHelper().InitBasic()
+	s.RunForSystemAdminAndLocal("system & local accounts can set a team to private", func(c client.Client) {
+		printer.Clean()
+		teamID := s.th.BasicTeam.Id
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("private", true, "")
+		err := modifyTeamsCmdF(c, cmd, []string{teamID})
+		s.Require().NoError(err)
```
Comment (mgdelacroix) [inline commands/team_e2e_test.go:64]: Could you please add an assertion for the returned team as well? Same suggestion applies for the other test cases

File: commands/team_e2e_test.go [hunk 5]
```diff
@@ -51,3 +52,48 @@ func (s *MmctlE2ETestSuite) TestRenameTeamCmdF() {
...
+		// teardown
+		appErr = s.th.App.UpdateTeamPrivacy(teamID, model.TEAM_OPEN, true)
+		s.Require().Nil(appErr)
+	})
+	s.Run("user that creates the team can't set team's privacy due to permissions", func() {
+		printer.Clean()
+		teamID := s.th.BasicTeam.Id
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("private", true, "")
+		s.th.LoginBasic()
```
Comment (mgdelacroix) [inline commands/team_e2e_test.go:79]: Login is done as part of the `s.SetupTestHelper().InitBasic()`, so unless you logout it should not be necessary to log back in

**Review by mgdelacroix — COMMENTED** [review]
Thanks for your contribution @jecepeda! Only one comment added, besides the pending ones from @ashishbhate

File: commands/team_e2e_test.go (hunk 1 above)
Comment (jecepeda) [reply commands/team_e2e_test.go:11]: Ok, thanks!!

File: commands/team_e2e_test.go (hunk 4 above)
Comment (jecepeda) [reply commands/team_e2e_test.go:67]: Hi @mgdelacroix where do I obtain the returned team?

**Review by ashishbhate — APPROVED** [review]

**Review by ashishbhate — APPROVED** [review]
Thanks @jecepeda ! LGTM

**Review by lindy65 — APPROVED** [review]
Thanks @jecepeda Looks good to me 👍

File: commands/team_e2e_test.go (hunk 4 above)
Comment (jecepeda) [reply commands/team_e2e_test.go:67]: Hey! I think I did it.

Thanks a ton for the suggestion!

File: commands/team.go [hunk 6]
```diff
@@ -343,10 +343,10 @@ func modifyTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 			printer.PrintError("Unable to find team '" + args[i] + "'")
 			continue
 		}
-		if _, response := c.UpdateTeamPrivacy(team.Id, privacy); response.Error != nil {
+		if updatedTeam, response := c.UpdateTeamPrivacy(team.Id, privacy); response.Error != nil {
 			printer.PrintError("Unable to modify team '" + team.Name + "' error: " + response.Error.Error())
 		} else {
-			printer.PrintT("Modified team '{{.Name}}'", team)
+			printer.PrintT("Modified team '{{.Name}}'", updatedTeam)
```
Comment (mgdelacroix) [inline commands/team.go:349]: <3

**Review by mgdelacroix — APPROVED** [review]
LGTM! Thanks a lot @jecepeda!! :confetti_ball:

**Pull request merged**